engine/api/src/main/java/org/apache/cloudstack/engine/subsystem/api/storage/VolumeService.java

```diff
@@ -92,7 +92,7 @@ public VolumeInfo getVolume() {
 
     AsyncCallFuture<VolumeApiResult> resize(VolumeInfo volume);
 
-    void resizeVolumeOnHypervisor(long volumeId, long newSize, long destHostId, String instanceName);
+    void resizeVolumeOnHypervisor(long volumeId, long currentSize, long newSize, long destHostId, String instanceName);
```

Comment:
why is the parameter be called `currentSize` needed? seems like `newSize` is enough.

Reply:
Yes it should suffice as currentsize can be got from volume. But in case of an uploaded volume with a custom disk offering specified, we set the new size (i.e., the size that it would get resized to eventually during attach) as the volume size, instead of the physical size. And since I am reusing this method to resize the volume after the attach operation, vol.size() = currentSize would be equal to newSize.

Reply:
I added the parameter because, for upload volume, currentSize would be equal to newSize (as both would point to volume.getSize()) if currentSize wasn't explicitly passed as the size of the volume on the secondary store.

Reply:
I still don't get why it has to be passed.

Reply:
So currently, https://github.com/apache/cloudstack/blob/9eefc76fc5ef9abf3d74a5cc6c528ca404c707fd/engine/storage/volume/src/main/java/org/apache/cloudstack/storage/volume/VolumeServiceImpl.java#L2429 - uses volume.getSize() as the current Size and newSize is what ever size we want the volume to get resized to. In case of resizing an attached volume, volume.getSize() would be the size to which volume would be resized to. And if I weren't to explicitly pass the actual size of the volume (as it was when on the secondary store) currentSize would be equal to newSize and hence resizing wouldn't happen. Because, with this PR I update the size of the uploaded volume to the size that the disk offering enforces https://github.com/apache/cloudstack/pull/6005/files#diff-b93da5c5293f8cb4470ff6b1642d312131bb053df07b71b4b951b08dff0ad9abR404-R405

Reply:
ok, thanks for explaining. I think this is a problem that needs to be addressed for all hypervisors separately. The old size should be retrieved from the hypervisor/primary storage. What we pass here could be off for any number of reasons. This is a bit out of scope here, but can you add a comment to what the problem is?